plugins/bundle/plugin.go

```diff
@@ -674,26 +673,26 @@ func (p *Plugin) saveBundleToDisk(name string, raw io.Reader) error {
 	return os.Rename(tmpFile, bundleFile)
 }
 
-func saveCurrentBundleToDisk(path, filename string, raw io.Reader) error {
+func saveCurrentBundleToDisk(path string, raw io.Reader) (string, error) {
 	if _, err := os.Stat(path); os.IsNotExist(err) {
 		err = os.MkdirAll(path, os.ModePerm)
 		if err != nil {
-			return err
+			return "", err
 		}
 	}
 
 	if raw == nil {
-		return fmt.Errorf("no raw bundle bytes to persist to disk")
+		return "", fmt.Errorf("no raw bundle bytes to persist to disk")
 	}
 
-	dest, err := os.OpenFile(filepath.Join(path, filename), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+	dest, err := os.CreateTemp(path, ".bundle.tar.gz.*.tmp")
```

Comment:
> [f](https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/os/tempfile.go;bpv=1;bpt=1;l=44?gsn=f&gs=kythe%3A%2F%2Fgo.googlesource.com%2Fgo%3Flang%3Dgo%3Fpath%3Dos%23var%2520%255Btempfile.go%252344%255D.f), [err](https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/os/tempfile.go;bpv=1;bpt=1;l=44?gsn=err&gs=kythe%3A%2F%2Fgo.googlesource.com%2Fgo%3Flang%3Dgo%3Fpath%3Dos%23var%2520%255Btempfile.go%252344%255D.err) := [OpenFile](https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/os/file.go;drc=846490110ab1117b5f7366e3a531d24d88800074;l=335)([name](https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/os/tempfile.go;drc=846490110ab1117b5f7366e3a531d24d88800074;l=43), [O_RDWR](https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/os/file.go;drc=846490110ab1117b5f7366e3a531d24d88800074;l=76)|[O_CREATE](https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/os/file.go;drc=846490110ab1117b5f7366e3a531d24d88800074;l=79)|[O_EXCL](https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/os/file.go;drc=846490110ab1117b5f7366e3a531d24d88800074;l=80), 0600)

this is what's called under the hood, so it's changed a bit -- but I think that's OK.